troll-o-matic.py

```diff
@@ -273,27 +284,53 @@ def process_changes(self, prefix, changes):
 
       gerrit_patch = rev.get_commit_from_remote('cros', cur_rev.ref)
 
-      upstream_shas = rev.get_cherry_pick_shas_from_patch(gerrit_patch)
-      if not upstream_shas:
-        self.handle_missing_hash_review(c, prefix)
-        continue
+      if prefix == 'FROMLIST':
+        upstream_patchworks = rev.get_am_from_from_patch(gerrit_patch)
+        if not upstream_patchworks:
+          self.handle_missing_am_review(c, prefix)
+          continue
 
-      upstream_patch = None
-      upstream_sha = None
-      for s in reversed(upstream_shas):
-        try:
-          upstream_patch = rev.get_commit_from_sha(s)
-          upstream_sha = s
-          break
-        except:
+        upstream_patch = None
+        for u in reversed(upstream_patchworks):
+          try:
+            upstream_patch = rev.get_commit_from_patchwork(u)
+            break
+          except:
+            continue
+
+        if not upstream_patch:
+          self.print_error(
+            'ERROR: patch missing from patchwork, or patchwork host '
+            'not whitelisted for {} ({})\n'.format(
+                                    c, upstream_patchworks))
+          self.blacklist.append(c)
+          continue
```

Comment:
How about pulling these out into functions:
`self.get_upstream_patch_from_patchwork()`
and
`self.get_upstream_patch_from_git()`

Reply:
done